MapboxCoreNavigation/RouteController.swift

```diff
@@ -611,35 +611,55 @@ extension RouteController: CLLocationManagerDelegate {
 
         for voiceInstruction in spokenInstructions {
             if userSnapToStepDistanceFromManeuver <= voiceInstruction.distanceAlongStep || firstInstructionOnFirstStep {
-
-                NotificationCenter.default.post(name: .routeControllerDidPassSpokenInstructionPoint, object: self, userInfo: [
-                    RouteControllerNotificationUserInfoKey.routeProgressKey: routeProgress
-                ])
-
+                
+                announcePassage(of: voiceInstruction, along: routeProgress)
                 routeProgress.currentLegProgress.currentStepProgress.spokenInstructionIndex += 1
+                
                 return
             }
         }
     }
     
     func updateVisualInstructionProgress() {
         guard let userSnapToStepDistanceFromManeuver = userSnapToStepDistanceFromManeuver else { return }
-        guard let visualInstructions = routeProgress.currentLegProgress.currentStepProgress.remainingVisualInstructions else { return }
+        guard let remainingVisualInstructions = routeProgress.currentLegProgress.currentStepProgress.remainingVisualInstructions else { return }
         
         let firstInstructionOnFirstStep = routeProgress.currentLegProgress.stepIndex == 0 && routeProgress.currentLegProgress.currentStepProgress.visualInstructionIndex == 0
         
-        for visualInstruction in visualInstructions {
+        for visualInstruction in remainingVisualInstructions {
             if userSnapToStepDistanceFromManeuver <= visualInstruction.distanceAlongStep || firstInstructionOnFirstStep {
                 
-                NotificationCenter.default.post(name: .routeControllerDidPassVisualInstructionPoint, object: self, userInfo: [
-                    RouteControllerNotificationUserInfoKey.routeProgressKey: routeProgress
-                    ])
-                
                 routeProgress.currentLegProgress.currentStepProgress.visualInstructionIndex += 1
+                announcePassage(of: visualInstruction, along: routeProgress)
+                
                 return
             }
         }
     }
+    
+    private func announcePassage(of spokenInstructionPoint: SpokenInstruction, along routeProgress: RouteProgress) {
+        
+        delegate?.router?(self, didPassSpokenInstructionPoint: spokenInstructionPoint, along: routeProgress)
+        
+        let info: [RouteControllerNotificationUserInfoKey: Any] = [
+                .routeProgressKey: routeProgress,
+                .spokenInstructionKey: spokenInstructionPoint
+        ]
+        
+        NotificationCenter.default.post(name: .routeControllerDidPassSpokenInstructionPoint, object: self, userInfo: info)
+    }
+    
+    private func announcePassage(of visualInstructionPoint: VisualInstructionBanner, along routeProgress: RouteProgress) {
```

Comment:
Should we tell the application that the user has passed the point for a VisualInstructionBanner or the VisualInstruction itself? There are situations where a tertiary banner can appear independently of (after) the same visual instruction’s primary and secondary banners; is this method called in that case?

Reply:
This is an interesting question, but I'm not sure it's in-scope of this PR, considering that I'm merely exposing the same objects we used to announce through notification, as a delegate message. 

Also, considering that the notification existed before this PR, there are probably deprecation concerns to think of.

Reply:
I brought up the issue of VisualInstructionBanner versus VisualInstruction because it might affect the signature of any public methods we introduce in this release regarding visual instructions. I think we need to figure out whether a tertiary instruction appearing independently, away from a maneuver, can trigger a notification and delegate method. If not, I suspect this method should take a VisualInstruction rather than a VisualInstructionBanner; likewise for the notification.

Reply:
This is sort of a confusing case -- the `VisualInstructionBanner` has a `VisualInstruction` as its `tertiaryInstruction`

Maybe there should be separate notifications if the case you cited is a thing we need to worry about. Is this something that happens frequently? Do you have an example handy?

Reply:
> I think we need to figure out whether a tertiary instruction appearing independently, away from a maneuver, can trigger a notification and delegate method.

@1ec5 it looks like this is [not the case](https://github.com/mapbox/mapbox-navigation-ios/pull/1912/files#diff-1a2149692b3d7c58b6a0392120d6011aR633). The only time `announcePassage(of:along:)` is called is when we approach a maneuver.

Reply:
I see, so NextBannerView knows to unhide itself not because of a specific notification but because it gets updated on every route progress update notification.